lib/commands.js

```diff
@@ -341,23 +343,29 @@ var operations = module.exports.operations = {
 
   promptParameters: function(context) {
     var newTemplateParameters = context.newTemplate.Parameters || {};
+    var overrideParameters = {};
 
     if (context.overrides.parameters) {
       Object.keys(context.overrides.parameters).forEach(function(key){
         if (newTemplateParameters[key] || context.oldParameters[key])
-          context.oldParameters[key] = context.overrides.parameters[key];
+          overrideParameters[key] = context.overrides.parameters[key];
```

Comment:
So the reason that we set the context object's old and new parameter values to the override value is so that parameter changes that are passed as overrides do not show up as part of the diff between old/new parameter sets. [This comparison ends up being done in the `confirmParameters` function](https://github.com/mapbox/cfn-config/blob/0cc06b5fadaf2c669874f07db04d4d62edfc538d/lib/commands.js#L379).

Could you keep this part that sets the override value in `context.oldParameters`, but collect all the TRUE old values in its own object so that you can pass the un-overriden content to `changesetParameters`?

Reply:
> So the reason that we set the context object's old and new parameter values to the override value is so that parameter changes that are passed as overrides do not show up as part of the diff between old/new parameter sets. [This comparison ends up being done in the `confirmParameters` function](https://github.com/mapbox/cfn-config/blob/0cc06b5fadaf2c669874f07db04d4d62edfc538d/lib/commands.js#L379).

I think we've already accounted for this.

In the [if/else clause](https://github.com/mapbox/cfn-config/pull/178/files#diff-5e05cf0e6359f91b6a5859876734d7f0L352-R368) right below here, we either:

- When `(context.overrides.force || context.overrides.skipPromptParameters)` is `true`, we **do not** assign the `overrideParameters` to `context.oldParameters`. This is the source of your concern. However, if you look at [`confirmParameters`](https://github.com/mapbox/cfn-config/pull/178/files#diff-5e05cf0e6359f91b6a5859876734d7f0R383-R388) you'll notice that it continues early on the same condition -- when `(context.overrides.force || context.overrides.skipPromptParameters)` is `true` -- and doesn't not perform the diff.
- When `(context.overrides.force || context.overrides.skipPromptParameters)` is `false`, we **DO** assign the `overrideParameters` to `context.oldParameters`. Then in `confirmParameters`, `context.oldParameters` has the expected values when the diff is performed.

Does this address your concern @rclark?

Reply:
Yeah, it does. Thanks @danmactough.